crates/quickjs-wasm-rs/src/js_binding/value.rs

```diff
@@ -187,8 +195,12 @@ impl<'a> JSValueRef<'a> {
 
     /// Converts the JavaScript value to a string if it is a string.
     pub fn as_str(&self) -> Result<&str> {
-        let buffer = self.as_wtf8_str_buffer();
-        str::from_utf8(buffer).map_err(Into::into)
+        if self.is_str() {
```

Comment:
This is a drive-by fix which I probably should have put in another PR. The function description says it only converts to a string if the JS value is a string, but the implementation never checked if it’s a string (as far as I can tell). This has led to some weird bugs for me where this function returned an `Ok` value, but didn’t contain anything.

Reply:
Removing this from the PR